r1cs-std/src/bits/boolean.rs

```diff
@@ -235,6 +235,16 @@ impl PartialEq for AllocatedBit {
 impl Eq for AllocatedBit {}
 
 impl<ConstraintF: Field> AllocGadget<bool, ConstraintF> for AllocatedBit {
+    fn alloc_constant<T, CS: ConstraintSystem<ConstraintF>>(
+        _cs: CS,
+        _t: T,
+    ) -> Result<Self, SynthesisError>
+    where
+        T: Borrow<bool>,
+    {
+        unimplemented!();
```

Comment:
Do we want to implement this via the `CS::one()` idea?

Reply:
Hmm, good point. We can just convert the bool to a bit and multiply by CS::one.

Reply:
This is going to be a bigger change than expected - it requires changing the `variable` field to a `ConstraintVar` which requires a type parameter, which will affect consumers. Any clever solution?

Reply:
hmm good point. I'll address this in the WIP constraint system refactor

Reply:
(Which will require adding the type parameter anyway)

Reply:
Alright :)